core/templates/domain/topic/editable-topic-backend-api.service.ts

```diff
@@ -16,233 +16,298 @@
-          successCallback({
-            topicDict: angular.copy(response.data.topic_dict),
-            groupedSkillSummaries: angular.copy(
-              response.data.grouped_skill_summary_dicts),
-            skillIdToDescriptionDict: angular.copy(
-              response.data.skill_id_to_description_dict),
-            skillQuestionCountDict: {
-              ...response.data.skill_question_count_dict
-            },
-            skillIdToRubricsDict: angular.copy(
-              response.data.skill_id_to_rubrics_dict),
-            classroomUrlFragment: response.data.classroom_url_fragment,
-          });
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
-      });
-    };
+import { AppConstants } from 'app.constants';
+import { BackendChangeObject } from 'domain/editor/undo_redo/change.model';
+import { TopicBackendDict } from 'domain/topic/TopicObjectFactory';
+import { TopicDomainConstants } from 'domain/topic/topic-domain.constants.ts';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service.ts';
 
-    var _fetchStories = function(
-        topicId, successCallback, errorCallback) {
-      var storiesDataUrl = UrlInterpolationService.interpolateUrl(
-        TOPIC_EDITOR_STORY_URL_TEMPLATE, {
-          topic_id: topicId
-        });
+interface FetchTopicBackendResponse {
+  'topic_dict': TopicBackendDict;
+  'grouped_skill_summary_dicts': Object;
+  'skill_id_to_description_dict': Object;
+  'skill_question_count_dict': Object;
+  'skill_id_to_rubrics_dict': Object;
+  'classroom_url_fragment': string;
+}
+
+interface FetchStoriesBackendResponse {
+  'canonical_story_summary_dicts': Object;
+}
+
+interface FetchSubtopicPageBackendResponse {
+  'subtopic_page': Object;
+}
+
+interface DeleteTopicBackendResponse {
+  status: Object;
+}
+
+interface UpdateTopicBackendResponse {
+  'topic_dict': Object;
+  'skill_id_to_description_dict': Object;
+  'skill_id_to_rubrics_dict': Object
+}
+
+interface DoesTopicWithUrlFragmentExistBackendResponse {
+  'topic_url_fragment_exists': Object
+}
 
-      $http.get(storiesDataUrl).then(function(response) {
-        var canonicalStorySummaries = angular.copy(
-          response.data.canonical_story_summary_dicts);
-        if (successCallback) {
-          successCallback(canonicalStorySummaries);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+interface DoesTopicWithNameExistBackendResponse {
+  'topic_name_exists': Object
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class EditableTopicBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService) {}
+
+  private _fetchTopic(
+      topicId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
```

Comment:
Please change 'Object' to the appropriate type here and elsewhere in the file.

Reply:
Done for all functions except _fetchTopic and _updateTopic. I am not sure what these two should be.

Reply:
Done